src/blocked.c

```diff
@@ -679,7 +679,7 @@ void handleClientsBlockedOnKeys(void) {
                 /* Edge case: If lookupKeyReadWithFlags decides to expire the key we have to
                  * take care of the propagation here, because afterCommand wasn't called */
                 if (server.also_propagate.numops > 0)
-                    propagatePendingCommands();
+                    afterDatasetChange();
```

Comment:
considering the rename, and also considering #9406 and #11310, i think we should drop the `if (server.also_propagate.numops > 0)`, we can move it inside the function if needed.

Reply:
@oranagra I prefer to do those changes on a follow up PR so this PR will be safer for cherrypick.

Reply:
Dropped